big_int.cpp

```diff
@@ -0,0 +1,519 @@
+       else if(a.sign ==1){ a.sign=0;// cout<< "b-a"<<endl;
+        return b-a;
+     }
+   else if(b.sign ==1){ b.sign=0;//cout<< "a-b"<<endl;
+    return a-b;
+ }
+      
+          int carry=0;
+          string str1=a.str;
+          string str2=b.str;
+          string result="";
+          //int l=max(str1.size(),str2.size());
+          int i=str1.size()-1;
+          int j=str2.size()-1;
+         for(;i>=0 || j>=0;)
+          {
+            
+             if (i>=0) {carry += str1[i]-'0'; }
+          if (j>=0) {carry += str2[j]-'0'; }
+              //string s=to_string(carry%10);
+              result.insert(0,1,(carry%10)+'0');
+              carry =carry/10;
+              //cout<<carry<<endl;
+              i--;
+              j--;
+            }
+            if(carry)
+
+            { //string s=to_string(carry);
+              result.insert(0 ,1,carry+'0');
+                    }
+          ans.str=result;
+          //cout<<ans.last;
+        //cout<<endl;
+           return ans;
+         }
+            
+              
+
+BigInteger operator- (BigInteger  a, BigInteger  b){
+       BigInteger ans;
+        if(a.sign==1 && b.sign==1){a.sign=0;b.sign=0; //cout<< "b-a"<<endl;
+        return b-a; }
+       else if(a.sign ==1){ a.sign=0;  ans=b+a; ans.sign=1;//cout<< "-(b+a)"<<endl; 
+       return ans; }
+      else if(b.sign ==1){ b.sign=0;//cout<< "b+a"<<endl; 
+      return a+b;}
+        else if(a<b){//cout<<"- b-a"; 
+        ans=b-a; ans.sign=1; return ans;}
+          
+          
+          int carry=0;
+         string str1=a.str;
+          string str2=b.str;
+          string result="";
+         int i=str1.size()-1;
+          int j=str2.size()-1;
+         for(;i>=0 || j>=0;)
+          {
+            
+             if (i>=0)  {carry += str1[i]-'0';   }
+           if (j>=0)  {carry -= str2[j]-'0';  }
+          //cout<<carry<<endl;
+                 if(carry<0)
+            { //string s=to_string((carry+10));
+             result.insert(0,1,(carry+10)+'0');
+             //cout<<carry+10<<endl;
+              carry=-1;
+              //cout<<carry<<endl;
+                 }
+              
+              else{ //string s=to_string((carry));
+                result.insert(0,1,carry+'0');
+                //ans.number[i]=(carry);
+             //  cout<<carry<<endl;
+              carry =0;
+              
+            }
+            i--;j--;
+            }
+            while(result[0]=='0' && result.length()!=1) 
+    result.erase(0,1);
+             ans.str=result;
+            return ans;
+        }
+
+      
+BigInteger operator* (BigInteger  a, BigInteger  b) {
+        //string s="";
+        
+     //   if(a.sign==1 && b.sign==1){ }
+     //  else if(a.sign ==1){ ans.sign=1;}
+     // else if(b.sign ==1){ ans.sign=1;}
+     string str1=a.str;
+          string str2=b.str;
+          
+            BigInteger ans;
+      int first;
+          int i,j;
+          int carry;
```

Comment:
Codacy found an issue: [The scope of the variable 'carry' can be reduced.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6290647)